### PR TITLE
Include Mandatory Questions

### LINKED ISSUE #506: Include mandatory questions

### PR DESCRIPTION
**Pull request by gnarbaiz**
closes #506

### CODE REVIEW

File: app/src/myanmar/java/org/eyeseetea/malariacare/database/PostMigration.java [hunk 1]
```diff
@@ -12,4 +12,5 @@
     public static void launchPostMigration() {
         Migration15AddOptionAttributeColumns.postMigrate();
     }
+
```
Comment (ifoche) [inline app/src/myanmar/java/org/eyeseetea/malariacare/database/PostMigration.java]: @gnarbaiz why this space? If you're not doing anything else in this file, remember 2 things:
- 1) making changes to any file increases the risk of conflict with other developers, so unless you really want to make a change, please don't do it
- 2) you have to review every file added for commit before committing them and even more before uploading them to github repository. Please only submit what's part of your solution

File: app/src/myanmar/res/layout/row_header_item.xml [hunk 2]
```diff
@@ -2,19 +2,39 @@
...
+        <org.eyeseetea.malariacare.views.TextCard
+            android:layout_width="wrap_content"
+            android:layout_height="match_parent"
+            android:layout_toRightOf="@+id/row_header_text"
+            android:visibility="visible"
+            android:layout_gravity="center_horizontal"
+            android:textColor="@color/red"
+            android:id="@+id/row_header_compulsory"
+            android:inputType="textLongMessage"
+            android:text="*"
```
Comment (ifoche) [inline app/src/myanmar/res/layout/row_header_item.xml]: If you look at my second comment on #506 you'll see that I attached you the image from the designer. You remarked during our last meeting that the asterisk looked more like an image than like a real asterisk. Does your red asterisk look like that one? Maybe you need to change this TextCard by an Image? What do you think?

**Comment by gnarbaiz** [comment]
@ifoche pull request is updated, you can proceed with merge.

File: app/src/main/java/org/eyeseetea/malariacare/layout/adapters/survey/DynamicTabAdapter.java [hunk 3]
```diff
@@ -716,6 +716,13 @@ public View getView(int position, View convertView, ViewGroup parent) {
                     showOrHideChildren(screenQuestion);
                     break;
             }
+            if(screenQuestion.getCompulsory()!=null && screenQuestion.getCompulsory() == 0) {
+                ImageView rowCompulsoryView = ((ImageView) tableRow.findViewById(R.id.row_header_compulsory));
```
Comment (ifoche) [inline app/src/main/java/org/eyeseetea/malariacare/layout/adapters/survey/DynamicTabAdapter.java]: @gnarbaiz please note this line breaks Laos & Cambodia variants. You cannot call from main an id that's only present on one variant. You need to move that part to a function only present on myanmar variant, so only myanmar will see the "R.id.row_header_compulsory" search.

**Comment by idelcano** [comment]
@ifoche  It's done, but with a simple validation (has value or not).
The "short text component" is validate with value "" , need we add a minimal or a limit?

**Comment by idelcano** [comment]
@ifoche  it's done again, i test better the migration and fix some problems.

**Pull request merged**